### ISSUE TITLE
Enable COI for desktop with `credentialless `

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#186614]
For desktop we can enable COI with `COEP: credentialless`. That's possible because chromium supports this new/proposed header value which relaxed loading cross-origin resources lacking the CORP header.

**Comment by jrieken** [comment]
Reverted, reopened because this has very likely caused https://github.com/microsoft/vscode/issues/187030

**Comment by jrieken** [comment]
Enabling COI caused a perf regression (https://github.com/microsoft/vscode/issues/187030) and therefore we are holding back. @deepak1556 Any idea why enabling COI would cause this?

**Comment by deepak1556** [comment]
Looked into this today and I have an answer for the slowdown,

In all cases, the first navigation to an origin happens from `about:blank`. So chromium creates a render process for the `null` origin and if the destination origin in our case `vscode-file://vscode-app` can commit in this process (always the case for first navigation) then the process will be reused without any browserinstance swap. This is what we see when coi is disabled, 

**BrowserInstance Swap result with COI disabled:**
![COI_disabled_swap_result](https://github.com/microsoft/vscode/assets/964386/57a90df4-3be7-41de-b90a-8a3dceec5ca9)

**Process list result with COI disabled:**
![Process_list_with_COI_disabled](https://github.com/microsoft/vscode/assets/964386/c40c751a-b47c-4c6e-9b9d-f593cf977d01)


When coi is enabled on an origin, chromium has logic (added in https://chromium-review.googlesource.com/c/chromium/src/+/3945190) to always force swap the browserinstance which leads to creation of new render process. This seems to apply to even when navigating from the `null` origin. Following screenshots show that there is a force swap on the browserinstance and there are two renderer process, pid 17824 from the `null` origin and pid 14756 for the `vscode-file://vscode-app` origin.

**BrowserInstance Swap result with COI enabled:**
![swap_result_coi_enabled](https://github.com/microsoft/vscode/assets/964386/e53eb139-3a1c-4132-89bc-d071d27e809e)

**Process list result with COI enabled:**
![process_list_coi_enabled](https://github.com/microsoft/vscode/assets/964386/0f50f15b-be00-4258-a997-20880321333a)

**RenderFrameHostImpl::DidCommitProvisionalLoad for COI enabled**
![didcommit_coi_enabled](https://github.com/microsoft/vscode/assets/964386/18f7fb32-4b67-40eb-a818-0ae713736da3)

It looks like the site instances now have a new lock attribute called `process_lock` which is `{}` for normal navigations, but ends up being `{ vscode-file://vscode-app/ cross-origin-isolated coi-origin='vscode-file://vscode-app' }` in the case of coi origin, so a coi origin does not get committed to a process that previously loaded a non coi origin.

tl:dr; there is a restart of the render process when coi is enabled and this is a penalty we will have to pay with coi origins for the initial navigation. Maybe chromium can adjust if first navigation on the app launch is to a coi origin then the process lock on null origin can be retagged and reused ? This needs discussion in upstream.

**Comment by deepak1556** [comment]
Upstream issue filed https://issues.chromium.org/issues/375314974

**Comment by DanTup** [comment]
@deepak1556 is this enabled now? I'm seeing wasm load in some Dart DevTools views where it didn't before (which I believe was because of no COI) and https://github.com/microsoft/vscode/pull/231665 looks like it enables this, but since this issue is still open and you linked a new Chromium issue that says "we are looking to enable coi", it's not clear to me what the current status is expected to be.

**Comment by deepak1556** [comment]
Hi @DanTup , nope it is not enabled yet (it was reverted in https://github.com/microsoft/vscode/pull/231899). The chromium issue to improve startup performance needs to be resolved first before we can adopt COI again. Upstream is working on it.

**Comment by eyebrowsoffire** [comment]
I mentioned this in the other bug, but this is because on the flutter engine side we added a single-threaded version of the wasm build that works in a non-COI context.
